spec/controllers/conference_registration_controller_spec.rb

```diff
@@ -0,0 +1,284 @@
+
+      it 'signs in registration user' do
+        expect(controller.current_user).not_to be_nil
+      end
+
+      it 'shows success message in flash notice' do
+        expect(flash[:notice]).to match('You are now registered and will be receiving E-Mail notifications.')
+      end
+
+      it 'redirects to registration show path' do
+        expect(response).to redirect_to conference_conference_registrations_path(conference_with_open_registration.short_title)
+      end
+
+      it 'creates a new registration' do
+        expect(Registration.count).to eq 1
+      end
+    end
+  end
+
+  context 'user is signed in' do
+    before { sign_in(user) }
+
+    describe 'GET #new' do
+      context 'user is registered to conference' do
+        before do
+          create(:registration, user: user, conference: conference_with_open_registration)
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'redirects to edit registration page' do
+          expect(response).to redirect_to edit_conference_conference_registrations_path(conference_with_open_registration.short_title)
+        end
+      end
+
+      context 'registration limit has reached' do
+        before do
+          conference_with_open_registration.update_attributes(registration_limit: 1)
+          create(:registration, conference: conference_with_open_registration)
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'redirects to root path' do
+          expect(response).to redirect_to root_path
+        end
+
+        it 'shows error in flash message' do
+          expect(flash[:alert]).to match "Sorry, registration limit exceeded for #{conference_with_open_registration.title}"
+        end
+      end
+
+      context 'successful request' do
+        before do
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'assigns registration and user variables' do
+          expect(assigns(:registration)).to be_instance_of(Registration)
+          expect(assigns(:user)).to be_instance_of(User)
+        end
+
+        it 'renders the new template' do
+          expect(response).to render_template('new')
+        end
+      end
+    end
+
+    describe 'GET #show' do
+      before do
+        @registration = create(:registration, conference: conference_with_open_registration, user: user)
+      end
+
+      context 'user has purchased a ticket' do
+        before do
+          @ticket = create(:ticket, conference: conference_with_open_registration)
+          @purchased_ticket = create(:ticket_purchase, conference: conference_with_open_registration,
+                                                       user: user,
+                                                       ticket: @ticket)
+          get :show, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'renders the show template' do
```

Comment:
I think you are suggesting that it is a common thing for both `tickets available` and `tickets not available`. How do I introduce this common context tho?